devel/query.lua

```diff
@@ -10,92 +10,61 @@ local validArgs = utils.invert({
+ 'maxlength',
+ 'excludetype',
+ 'excludekind',
 
- 'safer',
+ 'noblacklist',
  'dumb',
+
+ 'setvalue',
+ 'oneline',
  'disableprint',
- 'debug'
+ 'debug',
+ 'debugdata'
 })
 local args = utils.processArgs({...}, validArgs)
-depth=nil
-keydepth=nil
-cur_depth = -1
-cur_keydepth = -1
-
-newkeyvalue=nil
-bprintfields=nil
-bprintkeys=nil
-space_field="   "
-fN=0
---kN=25
+local new_value = nil
+local find_value = nil
+local maxdepth = nil
+local cur_depth = -1
+local bool_flags = {}
+
 local help = [====[
 
 devel/query
 ===========
 Query is a script useful for finding and reading values of data structure fields.
-Purposes will likely be exclusive to writing lua script code.
-
-Fields: contents of tables
-Keys:   contents below non-table containers
-
-Keys and fields are essentially the same thing. The real difference is what
-code checks need to be in place for queries on keys vs fields. This is why
-they are differentiated.
-
-This is a recursive script which takes your selected data {table,unit,item,tile}
-and then iterates through it, then iterates through anything it finds. It does
-this recursively until it has walked over everything it is allowed. Everything
-it walks over it checks against any (optional) string/value queries, and if it
-finds a match it then prints it to the console.
-
-You can control most aspects of this process, the script is fairly flexible. So
-much so that you can easily create an infinitely recursing query and/or potentially
-crash Dwarf Fortress and DFHack. In previous iterations memory bloat was even a
-concern, where RAM would be used up in mere minutes or seconds; you can probably
-get this to happen as well if you are careless with the depth settings and don't
-print everything walked over (i.e. have a search term). The `kill-lua` command
-may be able to stop this script if it gets out of control.
-
-Before recursing or printing things to the console the script checks several things.
-A few important ones:
-
- - Is the data structure capable of being iterated?
- - Is the data structure pointing to a parent data structure?
- - Is the current level of recursion too high, and do we need to unwind it first?
- - Is the number of entries too high (eg. 70,000 table entries that would be printed)?
- - Is the data going to be usefully readable?
- - Does the field or key match the field or key query or queries?
- - Is printing fields allowed?
- - Is printing keys allowed?
+Purposes will likely be exclusive to writing lua script code, possibly C++.
+
+This script takes your data selection eg.{table,unit,item,tile}
+then recursively iterates through it outputting names and values of what it finds.
+
+As it iterates you can have it do other things, like search for a specific structure pattern (see lua patterns)
+or set the value of fields matching the selection and any search pattern specified.
+
+If the script is taking too long to finish, or if it can't finish you should run ``dfhack-run kill-lua`` from a terminal.
```

Comment:
Since you actually print this out to the console, be mindful of the 80 column standard for help text

Reply:
Alright, I'll do that. But the standard should really be increased to 120 or more. Who in 2021 keeps their consoles at 80 wide.

Reply:
For that matter, the output of this command will not be pretty if the console is only 80 wide. So I feel as though maybe I shouldn't even commit changes to the help message.